File: leetcode/easy/Q206_ReverseLinkedList.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        if not head:
            return head

        nodes = []
        temp = head
        while temp is not None:
            nodes.append(temp)
            temp = temp.next

        nodes.reverse()
        length = len(nodes)
        for i in range(1, length):
            nodes[i-1].next = nodes[i]
        nodes[length-1].next = None
        return nodes[0]

File: src/easy/Q206_ReverseLinkedList.java
package easy;

import java.util.Stack;

public class Q206_ReverseLinkedList {
    class ListNode {
        int val;
        ListNode next;
        ListNode(){}
        ListNode(int x) {
            val = x;
            next = null;
        }
    }
    class Solution {
        public ListNode reverseList(ListNode head) {
            Stack<ListNode> stack = new Stack<>();
            ListNode node = head;
            while(node != null){
                stack.push(node);
                node = node.next;
            }

            node = new ListNode();
            head = node;
            while (!stack.isEmpty()){
                node.next = stack.pop();
                node = node.next;
            }
            node.next = null;
            return head.next;
        }
    }
}
